crates/common/tedge_config/tests/test_tedge_config.rs

```diff
@@ -753,16 +753,14 @@ fn test_parse_invalid_toml_file() -> Result<(), TEdgeConfigError> {
         "#;
 
     let (_tempdir, config_location) = create_temp_tedge_config(toml_conf)?;
-    let toml_path = config_location.tedge_config_file_path().to_string();
     let result = TEdgeConfigRepository::new(config_location).load();
 
     match result {
         Err(error @ TEdgeConfigError::Figment(_)) => {
-            assert_eq!(
-                error.to_string(),
-                format!(
-                    "unexpected character found: `<` at line 2 column 9 in {toml_path} TOML file"
-                )
+            assert!(
+                error.to_string().contains("invalid key"),
+                "error: {} does not contain the expected text: invalid key",
+                error
```

Comment:
Why the error returned by tedge_config has been changed?

Looking at the tested file content (`<abcde>`), the previous error message makes sense but not the new one.

Reply:
The error message changed in the latest version of the `figment` crate. So, I was forced to update this test. I personally feel that asserting exact error messages is a bit fragile due to issues like this. But didn't want to remove the whole assertion either, so decided to assert on just the significant part of the error message.